### ISSUE TITLE
Scheduled Jobs triggering on non-existant Database

### ISSUE THREAD

**Issue by patricktrainer** [metabase/metabase#11813]
Hello!

## The Issue
Im currently working through an issue that seems to reoccur on some sort of schedule. The result of the issue essentially locks Metabase's backend and makes running any query excruciatingly slow. 

## What Ive done
After doing some digging, Ive found a common link of 17 queries that have been running for hours. Killing these queries (by restarting the app server) immediately fixes the problem. 

## Metabase troubleshooting page
Poking through the troubleshooting page via the Admin page, I noticed some strangeness with some of the Jobs. Specifically, the `Triggers for metabase.task.sync-and-analyze.job` and `metabase.task.update-field-values.job`. 

In my case, "Database 4" seems culprit. 

## Non-existing db
To figure out what db "Database 4" mapped to, I checked the application db. I was surprised to find that there was no database where `id = 4`. 

## Next steps
Im at a loss of where to go at this point - any help would be greatly appreciated! 

### Screenshots
<img width="1597" alt="Screen Shot 2020-01-28 at 9 27 30 AM" src="https://user-images.githubusercontent.com/36834097/73279969-7a98b100-41b3-11ea-937a-7a119148229f.png">
<img width="495" alt="Screen Shot 2020-01-28 at 9 28 42 AM" src="https://user-images.githubusercontent.com/36834097/73279983-7ff5fb80-41b3-11ea-919f-fd5d206738cc.png">

## Application info
```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0",
    "vendor": ""
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.3+7",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.3",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.3+7",
    "os.name": "Linux",
    "os.version": "4.14.97-74.72.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "mysql",
      "redshift",
      "h2"
    ],
    "hosting-env": "elastic-beanstalk",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "date": "2019-12-19",
      "tag": "v0.34.0",
      "branch": "release-0.34.x",
      "hash": "1335215"
    },
    "settings": {
      "report-timezone": "US/Central"
    }
  }
}
```

**Comment by dacort** [comment]
Interesting - will have to see if we can reproduce this one. 

Do you know if you deleted a database around the time the trigger started? (2019-12-20)

**Comment by patricktrainer** [comment]
It's definitely possible @dacort. 
A database _was_ deleted, I cant recall if that's when the trigger initially started however, though.

**Comment by flamber** [comment]
Hi @patricktrainer 
I've tried various ways, but I cannot reproduce this issue. Tried to delete a database during the sync+scan process, and multiple other ways.

I'm fairly sure that the lockup you experienced was solved in 0.35.0 with the new connection handling. We saw several other instances that could lockup or leave hanging connections, which was solved in that version.

But I'm at a loss on how the database was removed from `metabase_database`, while it still left traces in the task/jobs (`task_history` and `qrtz_triggers`).
Could anyone have deleted the database directly in the database table instead of using the interface/API?

Or perhaps this happened during a lockup/hanging connection, which somehow caused the corruption? Looking at the code, I would have imagined that there would be several errors in the log, but I don't expect you have those logs stored somewhere?
